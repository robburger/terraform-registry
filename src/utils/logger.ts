export interface ILogger {
  trace(message: string, ...supportingData: unknown[]): void
  debug(message: string, ...supportingData: unknown[]): void
  info(message: string, ...supportingData: unknown[]): void
  warn(message: string, ...supportingData: unknown[]): void
  error(message: string, ...supportingData: unknown[]): void
}

export class Logger implements ILogger {
  /**
   * Prints to `stderr` the message and stack trace to the current position in the code.
   */
  public trace(message: string, ...supportingData: unknown[]): void {
    this.emitLog('trace', message, supportingData)
  }

  /**
   * Prints to `stdout` the message and any supporting data.
   */
  public debug(message: string, ...supportingData: unknown[]): void {
    this.emitLog('debug', message, supportingData)
  }

  /**
   * Prints to `stdout` the message and any supporting data.
   */
  public info(message: string, ...supportingData: unknown[]): void {
    this.emitLog('info', message, supportingData)
  }

  /**
   * Prints to `stderr` the message and any supporting data.
   */
  public warn(message: string, ...supportingData: unknown[]): void {
    this.emitLog('warn', message, supportingData)
  }

  /**
   * Prints to `stderr` the message and any supporting data.
   */
  public error(message: string, ...supportingData: unknown[]): void {
    this.emitLog('error', message, supportingData)
  }

  private emitLog(level: 'trace' | 'debug' | 'info' | 'warn' | 'error', message: string, supportingData: unknown[]) {
    if (supportingData.length > 0) {
      console[level](message, supportingData)
    } else {
      console[level](message)
    }
  }
}
