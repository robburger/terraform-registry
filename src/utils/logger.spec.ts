import { Logger } from './logger'

const logger = new Logger()

describe('custom logger', () => {
  test('create the logger and return it', () => {
    expect(logger).toMatchSnapshot()
  })
})
