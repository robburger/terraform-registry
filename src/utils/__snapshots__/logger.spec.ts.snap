// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`custom logger create the logger and return it 1`] = `Logger {}`;
